### DISCUSSION TITLE
Replace reflection with code generation

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/runtime#89530] [Ideas]
We can replace reflection actual behaviors with codegeneration?
To generate codes at runtime and improve performance in many scenarios
And help for native aot implementations

**Comment by Clockwork-Muse** [comment]
> To generate codes at runtime and improve performance in many scenarios

Code generation (available via the Code generators feature) happens at compile time.  It will not be available at runtime.
It definitely won't be available after AOT has been performed.

You may not need reflection, as in many cases wiring things together can be performed at runtime, assuming you've gathered the required info at compile time.

**Reply by danmoseley** [reply]
@sajjadarashhh  can you describe what you are trying to do? What are you reflecting over and do you know its shape at compile time?

**Comment by huoyaoyuan** [comment]
> To generate codes at runtime and improve performance in many scenarios

This is what reflection emit provides. Frameworks care about performance are already using it for a long history.

However, generating code is definitely more expensive than only using it once.
